src/Akka.Persistence.MongoDb/Snapshot/MongoDbSnapshotStore.cs

```diff
@@ -158,49 +158,37 @@ protected override async Task SaveAsync(SnapshotMetadata metadata, object snapsh
             var snapshotCollection = await GetSnapshotCollection(unitedCts.Token);
             
             var snapshotEntry = ToSnapshotEntry(metadata, snapshot);
-            await MaybeWriteWithTransaction(async (session, token) =>
-            {
-                if (session is not null)
-                {
-                    await snapshotCollection.ReplaceOneAsync(
-                        session: session,
-                        filter: CreateSnapshotIdFilter(snapshotEntry.Id),
-                        replacement: snapshotEntry,
-                        options: new ReplaceOptions { IsUpsert = true }, 
-                        cancellationToken: token);
-                }
-                else
-                {
-                    await snapshotCollection.ReplaceOneAsync(
-                        filter: CreateSnapshotIdFilter(snapshotEntry.Id),
-                        replacement: snapshotEntry,
-                        options: new ReplaceOptions { IsUpsert = true }, 
-                        cancellationToken: token);
-                }
-            }, unitedCts.Token);
+            
+            // According to
+            // https://www.mongodb.com/docs/manual/core/write-operations-atomicity/#atomicity-and-transactions
+            // A MongoDb write or update to a single document is guaranteed to be atomic, we don't need to
+            // enforce transactions for these.
+            await snapshotCollection.ReplaceOneAsync(
+                filter: CreateSnapshotIdFilter(snapshotEntry.Id),
+                replacement: snapshotEntry,
+                options: new ReplaceOptions { IsUpsert = true }, 
+                cancellationToken: unitedCts.Token);
```

Comment:
Also an optimization path, only a single snapshot will be written at a time, so there is no need for a transaction.
Also, should we add a feature flag to allow users to force this to run under transaction?